cdf/cdf.js

```diff
@@ -0,0 +1,143 @@
+const { interpolate } = require('./functions');
+
+class Cdf {
+  /**
+   * @param {number[]} xs
+   * @param {number[]} ys
+   */
+  constructor(xs, ys) {
+    this.xs = xs;
+    this.ys = ys;
+
+    if (!this.validateSize(this.xs, this.ys)) {
+      const corrected = this.convertToNewLength(this.xs.length);
+      this.xs = corrected.xs;
+      this.ys = corrected.ys;
+    }
+
+    const sorted = this.order(this.xs, this.ys);
+    this.xs = sorted.xs;
+    this.ys = sorted.ys;
+  }
+
+  /**
+   * Order them to make sure that xs are increasing
+   * @param {number[]} xs
+   * @param {number[]} ys
+   * @return {{ys: number[], xs: number[]}}
+   */
+  order(xs, ys) {
+    const xsYs = xs.map((v, i) => ({ ys: ys[i], xs: v }));
+    const sorted = xsYs.sort((a, b) => {
+      if (a.xs > b.xs) return 1;
+      if (a.xs < b.xs) return -1;
+      return 0;
+    });
+
+    const XS = sorted.map(v => v.xs);
+    const YS = sorted.map(v => v.ys);
+
+    return { xs: XS, ys: YS };
+  }
+
+  /**
+   * @param {number[]} xs
+   * @param {number[]} ys
+   * @return {boolean}
+   */
+  validateSize(xs, ys) {
+    return xs.length === ys.length;
+  }
+
+  /**
+   * If xs=[1,2,3], and ys=[5,6,7],
+   * then findY(1) = 5, findY(3) = 7, findY(1.5) = 5.5
+   * @param x
+   * @return {*}
+   */
+  findY(x) {
+    let firstHigherIndex = this.xs.findIndex(X => X > x);
+    let lowerOrEqualXIndex = firstHigherIndex - 1;
+    let needsInterpolation = this.xs[lowerOrEqualXIndex] !== x;
+    if (needsInterpolation) {
+      return interpolate(
+        this.xs[lowerOrEqualXIndex],
+        this.xs[firstHigherIndex],
+        this.ys[lowerOrEqualXIndex],
+        this.ys[firstHigherIndex],
+        x
+      );
+    } else {
+      return this.ys[lowerOrEqualXIndex];
+    }
+  }
+
+  /**
+   * If xs=[1,2,3], and ys=[5,6,7],
+   * then findX(5) = 1, findX(7) = 3, findY(5.5) = 1.5
+   * This should do the same thing as `findY`, but for Y.
+   * @param y
+   * @return {*}
+   */
+  findX(y) {
+    let firstHigherIndex = this.ys.findIndex(Y => Y > y);
+    let lowerOrEqualIndex = firstHigherIndex - 1;
+    let needsInterpolation = this.ys[lowerOrEqualIndex] !== y;
+    if (needsInterpolation) {
+      // @todo: should we turn axes?
```

Comment:
Does this still need to be done? I assume it's a technical question, rather than a feature question (one for me)

Reply:
I am not sure that got you properly here

Reply:
I mean, can you explain why the ``todo`` is there? Is that a question for me, or something you intend to fix in this PR or next?